src/core/continuous.jl

```diff
@@ -283,3 +283,10 @@ function SciMLBase.reinit!(integ::AbstractODEIntegrator, u0::AbstractVector,
     set_deviations!(integ, Q0)
     reinit!(integ, integ.u; kwargs...)
 end
+
+successful_step(integ) = true
+successful_step(integ::AbstractODEIntegrator{Alg, IIP, S, T}) where {Alg <:AbstractSimpleDiffEqODEAlgorithm, IIP, S, T} = true
```

Comment:
These shouldn't check for `true`. They should check if `get_state(integ)` contains `Inf` or `NaN`.

Reply:
I'm afraid that wouldn't work. `step!` with the  simple integrator errors `dt<dtmin` before we could check for `Inf` or `NaN`

Reply:
Does it **error** or does it **warn**? In either case, that's unfortunate but oh well what can't we do. 

Feel free to devise a special version for `SimpleATsit` otherwise leave it `true`. 

At least `successful_step(integ)` should check what I said above, because this would work for e.g., discrete systems.

Reply:
It gives an error and stops the execution. Indeed, `successful_step` should work that way, also for the discrete ones. Maybe I'll do that too

Reply:
> At least successful_step(integ) should check what I said above, because this would work for e.g., discrete systems.

Added the checking (4th commit) and tested it with Stroboscopic map, Projected Integrator and discrete integrator. Since this is more general, I suggest moving this function from `continuous.jl`. Maybe.